src/mcp/client/streamable_http.py

```diff
@@ -74,6 +74,18 @@ class RequestContext:
     sse_read_timeout: timedelta
 
 
+class AuthTokenProvider(Protocol):
+    """Protocol for providers that supply authentication tokens."""
```

Comment:
Can we flesh out the docstring here a bit more? I.e. explain that this class is a base class that can be extended to provide custom behavior when authenticating from an MCP client to an MCP server?